sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOTest.java

```diff
@@ -2092,26 +2146,29 @@ public void testWriteTables() throws Exception {
             jobIdTokenView,
             WriteDisposition.WRITE_EMPTY,
             CreateDisposition.CREATE_IF_NEEDED,
+            sideInputs,
             new IdentityDynamicTables());
 
-    DoFnTester<KV<ShardedKey<String>, List<String>>,
-        KV<TableDestination, String>> tester = DoFnTester.of(writeTables);
-    tester.setSideInput(jobIdTokenView, GlobalWindow.INSTANCE, jobIdToken);
-    tester.getPipelineOptions().setTempLocation("tempLocation");
-    for (KV<ShardedKey<String>, List<String>> partition : partitions) {
-      tester.processElement(partition);
-    }
+    PCollection<KV<TableDestination, String>> writeTablesOutput =
+        writeTablesInput.apply(writeTables);
 
-    Map<TableDestination, List<String>> tempTablesResult = Maps.newHashMap();
-    for (KV<TableDestination, String> element : tester.takeOutputElements()) {
-      List<String> tables = tempTablesResult.get(element.getKey());
-      if (tables == null) {
-        tables = Lists.newArrayList();
-        tempTablesResult.put(element.getKey(), tables);
-      }
-      tables.add(element.getValue());
-    }
-    assertEquals(expectedTempTables, tempTablesResult);
+    PAssert.thatMultimap(writeTablesOutput)
+        .satisfies(
+            new SerializableFunction<Map<TableDestination, Iterable<String>>, Void>() {
+              @Override
+              public Void apply(Map<TableDestination, Iterable<String>> input) {
+                assertEquals(input.keySet(), expectedTempTables.keySet());
```

Comment:
<!--thread_id:cc_140598581_t; commit:7202328ef0e10895b7c80bb5d1ec39244b09c6b8; resolved:0-->
<!--section:context-quote-->
> **jkff** wrote:
> This seems equivalent to assertEquals(HashMultimap.create(input), expectedTempTables).

<!--section:body-->
Except that doesn't compile, since input is not a Multimap. Also even if it did, it's unclear to me whether equals() would properly handle unordered lists of values (I think Multimap might just delegate to the underlying collection type).

Reply:
You're right about the equality - then you could do assertEquals(HashMultimap.create(input), HashMultimap.create(expectedTempTables)) - for HashMultimap, equality is defined in the way you need. Or you could just change expectedTempTables to explicitly be a HashMultimap.